### DISCUSSION TITLE
Apple 2FA is now required.  How are you all implementing it within your CI/CD setup?

### DISCUSSION THREAD

**Discussion by git-great** [fastlane/fastlane#18342] [Q&A]
As the title stated: Apple 2FA is now required.  How are you all implementing it within your CI/CD setup?

I know you can generate a session using fastlane spaceauth... command and in theory, use it for up to 30 days, however, any time someone logs into the build account (frequently in our case as we are constantly adding the account to new ITC teams and you have to log into accept the invites...), it invalidates that session. 

Are there any better ways to handle this?

**Comment by tylerablake** [comment]
Thank you for starting this discussion! I'm very curious to hear other inputs on this as well. 

We made the switch to using the ASC (AppStore Connect) api key for everything, but it doesn't support push notification certificates, we have a lot of apps with push notifications so we generate/refresh certificates (if necessary) during the build process, so we're not sure the best way to proceed to resolve the 2FA issues.

**Reply by git-great** [reply]
I'm sure we will go with the App Store Connect API as well but we use the "produce" action and it's still not supported according to the docs at https://docs.fastlane.tools/app-store-connect-api/#supported-actionstools 

That's the only piece missing for us to proceed with the ASC API at this time.

**Reply by dinsen** [reply]
Hi @tylerablake 
See my answer about push here:
https://github.com/fastlane/fastlane/discussions/18333

**Reply by dinsen** [reply]
Hi @CovertLeopard 
There is no endpoint within the App Store Connect API to create new apps in App Store Connect. 
So Produce can’t be fully updated to support the API Key.

**Comment by dinsen** [comment]
In general my advice will be: Use the API Key for all it currently support, since this will be the future 👽

**Comment by git-great** [comment]
I don't suppose you know if Apple plans to implement that or not?

**Reply by dinsen** [reply]
No, but you could make a feature request to Apple here:

https://feedbackassistant.apple.com
